File: exercism_data/src/767.rb
class Phrase

	def initialize(text)
		@text = text
	end

	def word_count
		words = @text.downcase.split(/\W+/) # Split on non-word (letter, number, underscore)
		counts = {}
		words.each do |word|
			counts[word] ||= 0
			counts[word] += 1
		end
		counts
	end

end

File: assignments/python/wc/src/767.py
import string
def word_count(phrase):
    counts = {}
    phrase = ''.join((
        c.lower() for c in phrase if c not in string.punctuation))
    whole_words = phrase.split()
    single_words = set(whole_words)
    for word in single_words:
        counts[word] = len([w for w in whole_words if w == word])
    return counts
